src/main/java/core/basesyntax/service/RegistrationServiceImpl.java

```diff
@@ -5,10 +5,44 @@
 import core.basesyntax.model.User;
 
 public class RegistrationServiceImpl implements RegistrationService {
-    private final StorageDao storageDao = new StorageDaoImpl();
+    private static final int MIN_AGE = 18;
+    private static final int MIN_PASSWORD_AND_LOGIN_LENGTH = 6;
+    private final StorageDao storageDao;
+
+    public RegistrationServiceImpl() {
+        storageDao = new StorageDaoImpl();
+    }
 
     @Override
     public User register(User user) {
+        validUser(user);
+        return checkedUserForRegistration(user);
+    }
+
+    private User validUser(User user) throws RegistrationException {
+        if (user == null) {
+            throw new RegistrationException("User can't be null");
+        }
+        if (user.getLogin() == null) {
+            throw new RegistrationException("Login can't be null");
+        }
```

Comment:
Let's create three different methods: `validateLogin`, `validatePassword`, `validateAge`. 

Keep there both logic for null/empty checks and for boundaries. Call these three methods in your `register`. Then return the user from dao.

Reply:
Please follow @rovein advice. You can also move null-user check to a separate method. You can move logic related to user existence check to your 'validateLogin' method. Also let's return void in your validation-related methods